From fb45cb1128418904bb9310b934033d19ce5cc574 Mon Sep 17 00:00:00 2001
From: gongxiao-intel <xiaoyan.gong@intel.com>
Date: Tue, 22 Nov 2022 16:29:40 +0000
Subject: [PATCH 10/12] igc: fix can't transmit packet though others queues
 after enable launch time on queue 0.

Transmission windows need to be initiated on others queues when TSN mode is enabled.
---
 drivers/net/igc/igc_ethdev.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/net/igc/igc_ethdev.c b/drivers/net/igc/igc_ethdev.c
index f5a5b9bfeb..1248d48067 100644
--- a/drivers/net/igc/igc_ethdev.c
+++ b/drivers/net/igc/igc_ethdev.c
@@ -1003,6 +1003,15 @@ eth_igc_start(struct rte_eth_dev *dev)
 		IGC_WRITE_REG(hw, IGC_STQT(0), 0);
 		IGC_WRITE_REG(hw, IGC_ENDQT(0), NSEC_PER_SEC);
 
+		IGC_WRITE_REG(hw, IGC_STQT(1), 0);
+		IGC_WRITE_REG(hw, IGC_ENDQT(1), NSEC_PER_SEC);
+
+		IGC_WRITE_REG(hw, IGC_STQT(2), 0);
+		IGC_WRITE_REG(hw, IGC_ENDQT(2), NSEC_PER_SEC);
+
+		IGC_WRITE_REG(hw, IGC_STQT(3), 0);
+		IGC_WRITE_REG(hw, IGC_ENDQT(3), NSEC_PER_SEC);
+
 		txqctl |= 0x1;
 		IGC_WRITE_REG(hw, IGC_TXQCTL(0), txqctl);
 
-- 
2.25.1

